### PR TITLE
Feat/theme switcher

### PR DESCRIPTION
**Pull request by fredericvilcot**
This PR implements following feature: https://github.com/okp4/ui/issues/96

### CODE REVIEW

File: src/ui/atoms/theme/ThemeSwitcher.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import React, { useCallback, useEffect } from 'react'
+import * as Switch from '@radix-ui/react-switch'
+import MoonIcon from '../../../assets/icons/moonIcon.svg'
+import SunIcon from '../../../assets/icons/sunIcon.svg'
+import type { Theme, ThemeContextType } from './themeContext'
+import { useTheme } from 'hooks/useTheme'
+import './themeSwitcher.scss'
+
+export const ThemeSwitcher: React.FC = () => {
```
[the remark this answers is not in the document]
Comment (fredericvilcot) [reply src/ui/atoms/theme/ThemeSwitcher.tsx:9]: > Function `ThemeSwitcher` has 36 lines of code (exceeds 25 allowed). Consider refactoring.

@ccamel What can we do about codeclimate review which imposes a size limit even on React components ?

**Review by nperon — DISMISSED** [review]

**Review by ad2ien — APPROVED** [review]
:+1: tested by importing it in an other react project!

File: src/ui/atoms/theme/ThemeSwitcher.tsx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  const { theme, setTheme }: ThemeContextType = useTheme()
+  const isDarkTheme = theme === 'dark'
+
+  const saveTheme = useCallback(
+    (newTheme: Theme): void => {
+      setTheme(newTheme)
+      localStorage.setItem('okp4UserTheme', newTheme)
```
[the remark this answers is not in the document]
Comment (ccamel) [reply src/ui/atoms/theme/ThemeSwitcher.tsx:16]: @fredericvilcot I propose you to introduce a bunch of hooks to improve composability. Let's see this together.

**Review by ccamel — CHANGES_REQUESTED** [review]
Hey! What a good job. 👍 

I have some remarks and suggestions that we can look at together to finalize the PR. Thank you.

File: src/ui/atoms/canvas/Canvas.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import React, { useEffect, useRef } from 'react'
-import { useAnimationFrame } from 'hook/useAnimationFrame'
+import { useAnimationFrame } from 'hooks/useAnimationFrame'
```
[the remark this answers is not in the document]
Comment (fredericvilcot) [reply src/ui/atoms/canvas/Canvas.tsx]: @ccamel, I changed it, I didn't notice it ...

File: src/hooks/useTheme.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react'
+import ThemeContext from '../ui/atoms/theme/themeContext'
+import type { ThemeContextType } from '../ui/atoms/theme/themeContext'
```
[the remark this answers is not in the document]
Comment (fredericvilcot) [reply src/hooks/useTheme.ts]: @ccamel You're right, I created a folder dedicated to context, as `hooks` depend now on context and not ui ...

File: stories/atoms/brand/theme/Theme.stories.mdx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+  <ThemProvider>
+    <App />
+  <ThemeProvider>
+</React.StrictMode>
+```
+
+### Theme Switcher
+
+The `<ThemeSwitcher />` component can be invoked anywhere in your application.
+As an example, you could insert it in your header component to display it near toolbar actions :
```
[the remark this answers is not in the document]
Comment (fredericvilcot) [reply stories/atoms/brand/theme/Theme.stories.mdx]: 👀

**Comment by fredericvilcot** [comment]
@FYI @ErikssonJoakim

**Review by amimart — APPROVED** [review]

**Review by ad2ien — APPROVED** [review]
:clap: :clap: :clap:

**Pull request merged**